src/devtools/views/Components/Element.js

```diff
@@ -29,6 +30,8 @@ export default function ElementView({ index, style }: Props) {
     selectedElementID,
     selectElementByID,
   } = useContext(TreeContext);
+  const bridge = useContext(BridgeContext);
+  const store = useContext(StoreContext);
```

Comment:
I wonder how much it will impact performance to have every list item reading from multiple contexts? Should we leverage the `itemData` prop for this?

The below diff has a Flow error that I would need to sort out, but as an example:
```diff
diff --git a/src/devtools/views/Components/Element.js b/src/devtools/views/Components/Element.js
index 6fd7db8..91c8064 100644
--- a/src/devtools/views/Components/Element.js
+++ b/src/devtools/views/Components/Element.js
@@ -11,18 +11,20 @@ import React, {
 import { ElementTypeClass, ElementTypeFunction } from 'src/devtools/types';
 import { createRegExp } from '../utils';
 import { TreeContext } from './TreeContext';
-import { BridgeContext, StoreContext } from '../context';
 
 import type { Element } from './types';
+import type { ItemData } from './Tree';
 
 import styles from './Element.css';
 
 type Props = {
+  data: ItemData,
   index: number,
   style: Object,
 };
 
-export default function ElementView({ index, style }: Props) {
+export default function ElementView({ data, index, style }: Props) {
+  const { highlightElement } = data;
   const {
     baseDepth,
     getElementAtIndex,
@@ -30,8 +32,6 @@ export default function ElementView({ index, style }: Props) {
     selectedElementID,
     selectElementByID,
   } = useContext(TreeContext);
-  const bridge = useContext(BridgeContext);
-  const store = useContext(StoreContext);
 
   const element = getElementAtIndex(index);
 
@@ -60,6 +60,8 @@ export default function ElementView({ index, style }: Props) {
 
   // TODO Add click and key handlers for toggling element open/close state.
 
+  // Individual elements don't have a corresponding leave handler.
+  // Instead, it's implemented on the tree level.
   const handleMouseDown = useCallback(
     ({ metaKey }) => {
       if (id !== null) {
@@ -69,20 +71,12 @@ export default function ElementView({ index, style }: Props) {
     [id, selectElementByID]
   );
 
-  const rendererID = store.getRendererIDForElement(element.id) || null;
   // Individual elements don't have a corresponding leave handler.
   // Instead, it's implemented on the tree level.
-  const handleMouseEnter = useCallback(() => {
-    if (rendererID !== null) {
-      bridge.send('highlightElementInDOM', {
-        displayName: element.displayName,
-        id: element.id,
-        rendererID,
-        scrollIntoView: false,
-        isSticky: true,
-      });
-    }
-  }, [bridge, element, rendererID]);
+  const handleMouseEnter = useCallback(
+    () => highlightElement(((element: any): Element)),
+    [element, highlightElement]
+  );
 
   // Handle elements that are removed from the tree while an async render is in progress.
   if (element == null) {
diff --git a/src/devtools/views/Components/Tree.js b/src/devtools/views/Components/Tree.js
index ef6073e..af35102 100644
--- a/src/devtools/views/Components/Tree.js
+++ b/src/devtools/views/Components/Tree.js
@@ -12,14 +12,25 @@ import AutoSizer from 'react-virtualized-auto-sizer';
 import { FixedSizeList } from 'react-window';
 import { TreeContext } from './TreeContext';
 import { SettingsContext } from '../Settings/SettingsContext';
-import { BridgeContext } from '../context';
-import Element from './Element';
+import { BridgeContext, StoreContext } from '../context';
+import ElementView from './Element';
 import InspectHostNodesToggle from './InspectHostNodesToggle';
 import OwnersStack from './OwnersStack';
 import SearchInput from './SearchInput';
 
 import styles from './Tree.css';
 
+import type { Element } from './types';
+
+type HighlightElementFn = (element: Element) => void;
+
+export type ItemData = {|
+  baseDepth: number,
+  highlightElement: HighlightElementFn,
+  numElements: number,
+  getElementAtIndex: (index: number) => Element | null,
+|};
+
 type Props = {||};
 
 export default function Tree(props: Props) {
@@ -34,7 +45,8 @@ export default function Tree(props: Props) {
     selectPreviousElementInTree,
   } = useContext(TreeContext);
   const bridge = useContext(BridgeContext);
-  const listRef = useRef<FixedSizeList<any> | null>(null);
+  const store = useContext(StoreContext);
+  const listRef = useRef<FixedSizeList<ItemData> | null>(null);
   const treeRef = useRef<HTMLDivElement | null>(null);
 
   const { lineHeight } = useContext(SettingsContext);
@@ -93,15 +105,32 @@ export default function Tree(props: Props) {
     selectPreviousElementInTree,
   ]);
 
+  const highlightElement = useCallback<HighlightElementFn>(
+    (element: Element) => {
+      const rendererID = store.getRendererIDForElement(element.id);
+      if (rendererID !== null) {
+        bridge.send('highlightElementInDOM', {
+          displayName: element.displayName,
+          id: element.id,
+          rendererID,
+          scrollIntoView: false,
+          isSticky: true,
+        });
+      }
+    },
+    [bridge, store]
+  );
+
   // Let react-window know to re-render any time the underlying tree data changes.
   // This includes the owner context, since it controls a filtered view of the tree.
-  const itemData = useMemo(
+  const itemData = useMemo<ItemData>(
     () => ({
       baseDepth,
+      highlightElement,
       numElements,
       getElementAtIndex,
     }),
-    [baseDepth, numElements, getElementAtIndex]
+    [baseDepth, highlightElement, numElements, getElementAtIndex]
   );
 
   const handleMouseLeave = useCallback(() => {
@@ -128,7 +157,7 @@ export default function Tree(props: Props) {
               ref={listRef}
               width={width}
             >
-              {Element}
+              {ElementView}
             </FixedSizeList>
           )}
         </AutoSizer>
```

Maybe this isn't a big deal and we should just ignore it for now?